src/lib/auth.ts

```diff
@@ -502,17 +519,31 @@ export const createFakeAuthentication = (options: CreateFakeAuthenticationOption
     students: [],
     teachers: [],
   };
-  for (let i = 1; i <= NUM_FAKE_STUDENTS; i++) {
+  //This allows for student groups to be remembered when switching tabs or in cypress tests
+  if (parseInt(userId,10) <= NUM_FAKE_STUDENTS) {
+    for (let i = 1; i <= NUM_FAKE_STUDENTS; i++) {
```

Comment:
Ah this explains why my fake student ids that are greater than 99 were causing problems.